numpy/lib/tests/test_function_base.py

```diff
@@ -1447,6 +1448,38 @@ def test_size_zero_output(self):
         with assert_raises_regex(ValueError, 'new output dimensions'):
             f(x)
 
+    @pytest.mark.skipif(not HAS_REFCOUNT, reason="Python lacks refcounts")
+    def test_leaks(self):
+        # gh-11867
+        # np.frompyfunc(<class method>) creates a reference cycle
+        # that requires a gc collection cycle to break (on CPython 3)
```

Comment:
Have we worked out why this is the case? Am I right in thinking that this patch changes it from being unbreakable to just requiring a gc to break?

Reply:
Correct. I do not really understand the root cause and why only in this specific case